### ISSUE TITLE
Add CatBoost support

### ISSUE THREAD

**Issue by annaveronika** [mlflow/mlflow#2769]
It would be nice to add CatBoost, one of the three most popular gradient boosting libraries. (https://catboost.ai/, https://github.com/catboost/catboost). Both XGBoost and LightGBM are already supported.

**Comment by harupy** [comment]
https://github.com/mlflow/mlflow/pull/2417

**Comment by annaveronika** [comment]
Do you plan to merge it?

**Comment by EgorKraevTransferwise** [comment]
It would indeed be awesome if this was merged ;)

**Comment by norbert-liki** [comment]
I am looking forward to it also! ;)

**Comment by mareksimunek** [comment]
I am just considering using mlflow and catboost support would be great.
@harupy Hi, thanks for your effort.
 In what phase is your contribution? Are you waiting for review or need some help?

**Comment by harupy** [comment]
@EgorKraevTransferwise @norbert-liki @mareksimunek I have updated the PR.

**Comment by mareksimunek** [comment]
@harupy you are the best! I hope I will have time to try it this week.

Is it waiting for someone to review it or what's the process?

**Comment by harupy** [comment]
@mareksimunek 

> Is it waiting for someone to review it or what's the process?

Yes, it needs to be approved by an MLflow team member.

**Comment by coder46** [comment]
Looking forward for catboost support ! Thanks !

**Comment by mareksimunek** [comment]
@harupy would you please ping any of MLflow team member to look at it? By checking your contribution in this project seems like you know who to ask :).

**Comment by sephib** [comment]
Hi,
@harupy   - have you considered to support `autologging` ?

**Comment by harupy** [comment]
@sephib Yes, I think we could use the same approach as XGBoost and LightGBM.

**Comment by sephib** [comment]
@harupy Please let me know if I can assist in any manner

**Comment by pedrito87** [comment]
@harupy awesome that you've made that PR! Is there any plan to move forward with it? It's been some time now, and it would be awesome to have it merged :)

**Comment by harupy** [comment]
#2417 has been merged!
